### PR TITLE
Dev plus app conf

### PR DESCRIPTION
**Pull request by dataswifty**

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+include ".scalafmt-common.conf" 
+
+// ScalaFmt stuff
```
Comment (benshaw) [inline .scalafmt.conf:3]: Unless you are intentionally overwriting common i do not think this is required
if you are are overwriting common then id be happy to change common so it meets your needs

File: hat/app/org/hatdex/hat/api/controllers/DataDebits.scala [hunk 2]
```diff
@@ -48,140 +53,242 @@ class DataDebits @Inject() (
...
+    with RichDataJsonFormats {
 
   private val logger = loggingProvider.logger(this.getClass)
 
   def registerDataDebit(dataDebitId: String): Action[DataDebitSetupRequest] =
-    SecuredAction(WithRole(Owner(), DataDebitOwner(""), Platform()) || ContainsApplicationRole(Owner(), DataDebitOwner(""), Platform())).async(parsers.json[DataDebitSetupRequest]) { implicit request ⇒
-      dataDebitService.createDataDebit(dataDebitId, request.body, request.identity.userId)
-        .andThen(dataEventDispatcher.dispatchEventDataDebit(DataDebitOperations.Create()))
-        .map(debit ⇒ Created(Json.toJson(debit)))
+    SecuredAction(
```
Comment (augustinas) [inline hat/app/org/hatdex/hat/api/controllers/DataDebits.scala:63]: Oh, we'll need to look into scalafmt config for lines 63-68 refactor.
This

break

up 

is 

not

readable.
Not blocking the PR for this though.

**Comment by dataswifty** [comment]
> Wait... Shouldn't the PR go to the `dev` branch first? It currently merges directly into `staging`

It's is.  It's dev plus the config change. Marios said the change should be in staging and not dev.  

If there is no issue with the application being in dev, then I'll add it there and issue a new PR from dev to staging.

File: .scalafmt.conf (hunk 1 above)
Comment (dataswifty) [reply .scalafmt.conf:3]: Sure, would you make the change to remove the contents of scalafmt.conf please?

File: hat/app/org/hatdex/hat/api/controllers/DataDebits.scala (hunk 2 above)
Comment (dataswifty) [reply hat/app/org/hatdex/hat/api/controllers/DataDebits.scala:63]: I
Don't
See
The
Problem.  

If you can find the config to suppress this change I will add it.  

I don't mind either format. :)

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,7 +1,16 @@
 logLevel := Level.Warn
 
+// Typesafe Resolvers
 resolvers += Resolver.typesafeRepo("releases")
 
+// S3 based SBT resolver
+resolvers ++= Seq(
+  "HAT Library Artifacts Releases" at "https://s3-eu-west-1.amazonaws.com/library-artifacts-releases.hubofallthings.com",
+  "HAT Library Artifacts Snapshots" at "https://s3-eu-west-1.amazonaws.com/library-artifacts-snapshots.hubofallthings.com",
+  "HAT Library Artifacts Releases" at "https://s3-eu-west-1.amazonaws.com/library-artifacts-releases.hubofallthings.com"
```
Comment (benshaw) [inline project/plugins.sbt:10]: Releases should only occur once im not sure what happens if you have repeated sources

File: project/plugins.sbt [hunk 4]
```diff
@@ -10,20 +19,20 @@ addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.25")
...
 
 addSbtPlugin("org.hatdex" % "sbt-slick-postgres-generator" % "0.0.11")
 
 // run "sbt dependencyUpdates" to check maven for updates or "sbt ";dependencyUpdates; reload plugins; dependencyUpdates" for sbt plugins
 addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.3.4")
+
+// ScalaFMT
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.3.4")
+
+addSbtPlugin("io.dataswift" % "sbt-scalafmt-common" % "0.1.1-SNAPSHOT")
```
Comment (benshaw) [inline project/plugins.sbt:38]: use version 0.1.1 NOT SNAPSHOT

**Pull request merged**